### ISSUE TITLE
Refactor GraphChoice(s) and its associated property.

### ISSUE THREAD

**Issue by veillette** [phetsims/calculus-grapher#107]
Currently in `CalculusGrapherCommonScreenView.js`, we have the following types

```
type GraphType = 'original' | 'integral' | 'derivative' | 'secondDerivative';
export type GraphChoice = GraphType[];
export type GraphChoices = GraphChoice[];
```

and an associated property 

```
  protected readonly graphsSelectedProperty: Property<GraphChoice>;
```

In addition the buttonGroup associated with the graphChoice property is graphChoiceRadioButtonGroup.

- We should considering refactoring ` graphsSelectedProperty` to  better reflect the type of the property.

- There are some inconsistencies with the pluralization. `GraphChoices` is an array of array but only choice is pluralized.
 

My proposal is the following:

Since `GraphChoice` is an array of `GraphType`,  it may be better to call it `GraphsChoice`, and then to be consistent we would need to refactor `GraphChoices` to `GraphsChoices`, in which case the associated property would be `graphsChoiceProperty`  and the associated buttonGroup would be ` GraphsChoiceRadioButtonGroup`

@pixelzoom, do you have any thoughts or a better idea?

**Comment by pixelzoom** [comment]
Thinking about this a little more.   I think I would start by getting rid of  `GraphChoices/GraphSet` altogether, and simplify GraphChoice.ts to this:

```typescript
type GraphChoiceValues = 'original' | 'integral' | 'derivative' | 'secondDerivative';
export type GraphChoice = ( typeof GraphType )[ number ]
```

Then use `GraphChoice` and `GraphChoice[]` and `GraphChoice[][]` as appropriate, and see how that feels.  It is certainly more obvious.

**Comment by pixelzoom** [comment]
... but the problem with https://github.com/phetsims/calculus-grapher/issues/107#issuecomment-1317312694 is what to name things of type `GraphChoice[][]`:

```typeScript
graphChoice: GraphChoice
graphChoices: GraphChoice[]
??: GraphChoice[][]
```

**Comment by pixelzoom** [comment]
... but the problem with https://github.com/phetsims/calculus-grapher/issues/107#issuecomment-1317312694 is what to name variables of type `GraphChoice[][]`:

```typeScript
graphChoice: GraphChoice
graphChoices: GraphChoice[]
??: GraphChoice[][]
```

So I guess I'm still leaning towards:

```typeScript
type GraphChoiceValues = 'original' | 'integral' | 'derivative' | 'secondDerivative';
export type GraphChoice = ( typeof GraphType )[ number ];
export type GraphSet = GraphChoice[];
```

with variable names:

```typescript
graphChoice: GraphChoice
graphSet: GraphSet
graphSets: GraphSet[]
```

**Comment by pixelzoom** [comment]
Or maybe this isn't a typical use of string union, and I shouldn't be thinking of it as an enumation? 

And maybe I'm getting confused because natural-selection also has a `GraphChoice` enumeration (which is not a string union only for reasons of PhET-iO backward compatibility.)

Sorry to be so long winded.  Maybe we should discuss before making any major changes?

**Comment by pixelzoom** [comment]
OK, I think I have myself reoriented on this.  GraphChoice was not intended to be an enumeration.  So a simpler change is needed.  And I think it's mainly the GraphChoice name that's confusing.  So how about changing GraphChoice to something like GraphSet?  Like this:

```typescript
type GraphType = 'original' | 'integral' | 'derivative' | 'secondDerivative';
export type GraphSet = GraphType[];
```

And simply use `GraphSet[]` instead of adding `type GraphSets = GraphSet[]`.

**Comment by veillette** [comment]
@pixelzoom : give me the greenlight and I'll implement the changes above.

**Comment by pixelzoom** [comment]
What you've summarized in https://github.com/phetsims/calculus-grapher/issues/107#issuecomment-1317448376 sounds great to me.  If it were me,  I'd also move this bit into GraphSet.ts, but don't mind if you leave it in CalculusGrapherScreenView.ts:

```typescript
type GraphType = 'original' | 'integral' | 'derivative' | 'secondDerivative';
export type GraphSet = GraphType[];
```

**Comment by pixelzoom** [comment]
One other thing occurred to me.  GraphSet is indeed a "set", and therefore its values should be unique.  And if they are not unique, that's would be a problem, since only 1 instance of each graph exists.

We don't currently have a type for "unique array", but here's an article on how to implement one: https://ja.nsommer.dk/articles/type-checked-unique-arrays.html.  This might be a nice thing to add to phet-core/types/, and I'd be happy to take the on, if you think it's important.

For the time being, you can just add an assertion that verifies uniqueness, something like:

```js
assert && assert( graphSet.length === _.unique( graphSet ) )
```

**Comment by veillette** [comment]
I implemented the comment in https://github.com/phetsims/calculus-grapher/issues/107#issuecomment-1317448376

I also added an assertion to check that each graphSet has unique values.  

A second assertion should also be added to ensure that each graphSet in options.graphSets is unique. 
Maybe that warrants creating a new Class, or using a Javascript Set might help. I'll put in as a discussion point for our next meeting.
